assets/test/event_test.js

```diff
@@ -211,15 +225,74 @@ describe("pushEvent replies", () => {
     })
   })
 
-  test("pushEvent without connection noops", () => {
+  test("rejects with error", (done) => {
+    let view
+    let liveSocket = new LiveSocket("/live", Socket, {
+      hooks: {
+        Gateway: {
+          mounted(){
+            stubNextChannelReplyWithError(view, "error")
+            this.pushEvent("charge", {amount: 123}).catch((error) => {
+              expect(error).toEqual({error: "error"})
+              done()
+            })
+          }
+        }
+      }
+    })
+    view = simulateView(liveSocket, [], "")
+    view.update({
+      s: [`
+      <div id="gateway" phx-hook="Gateway">
+      </div>
+    `]
+    }, [])
+  })
+
+  test("pushEventTo - promise with multiple targets", (done) => {
+    let view
+    let liveSocket = new LiveSocket("/live", Socket, {
+      hooks: {
+        Gateway: {
+          mounted(){
+            stubNextChannelReply(view, {transactionID: "1001"})
+            this.pushEventTo("[data-foo]", "charge", {amount: 123}).then((result) => {
+              expect(result).toEqual([
+                {status: "fulfilled", value: {ref: 0, reply: {transactionID: "1001"}}},
+                // we only stubbed one reply
+                {status: "rejected", reason: expect.any(Error)}
```

Comment:
One difference between `pushEvent` and `pushEventTo` is that for pushEvent, we only resolve the `reply`, but here the fulfilled value is `{ ref, reply }`. We could map over the values and remove the ref if we want.